src/traces/bar/plot.js

```diff
@@ -469,12 +471,64 @@ function getTransform(textX, textY, targetX, targetY, scale, rotate) {
     return transformTranslate + transformScale + transformRotate;
 }
 
-function getText(trace, index) {
-    var value = helpers.getValue(trace.text, index);
+function getText(calcTrace, index, xa, ya) {
+    var trace = calcTrace[0].trace;
+
+    var value;
+    if(!trace.textinfo || trace.textinfo === 'none') {
+        value = helpers.getValue(trace.text, index);
```

Comment:
Hmm. `textinfo: 'none'` should mean no on-graph text even if `text` is set. In fact, `textinfo: 'none'` is essentially equivalent to `textposition: 'none'`, perhaps we don't need it.

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/3790/commits/bed68427ea138dfb004500581a39350a6eec041b.

Reply:
Please add jasmine test.

Reply:
Thanks!